langchain/document_loaders/fauna.py

```diff
@@ -0,0 +1,56 @@
+from typing import List, Optional
+from langchain.docstore.document import Document
+
+from langchain.document_loaders.base import BaseLoader
+
+
+class FaunaLoader(BaseLoader):
+    def __init__(
+        self,
+        query: str,
+        page_content_field: str,
+        secrect: str,
+        metadata_fields: Optional[List[str]] = None,
+        after: Optional[str] = None,
+    ):
+        self.query = query
+        self.page_content_field = page_content_field
+        self.secrect = secrect
+        self.metadata_fields = metadata_fields
+
+    def load(self) -> List[Document]:
```

Comment:
For a lazy implementation, we want to implement `lazy_load` rather than `load` -- take a look at the base class to see the method signatures with docs that explain what's going on! :)